### PR TITLE
【AMP Prim OP】support instance_norm prim ops for fp16 and bf16 dtype

### PR DESCRIPTION
**Pull request by Charles-hit**
### PR types

Others
### PR changes

Others
### Description

Pcard-66975
PR改动：

- 为`instance_norm`前反向拆解适配低精度规则，与原kernel保持一致，并且添加fp16，bfp16组合算子单测。
- 修复`instance_norm` 算子 `infermeta`数据类型推导错误的问题，导致组合算子在进行program检查会报类型不匹配。

### CODE REVIEW

**Review by cxxly — APPROVED** [review]
LGTM

**Review by Aurelius84 — APPROVED** [review]

File: test/legacy_test/test_instance_norm_op_v2.py [hunk 1]
```diff
@@ -12,15 +12,69 @@
...
+from paddle.fluid import Program, core, program_guard
+
+
+def instance_norm_wrapper(
+    input, weight, bias, epsilon=1e-5, momentum=0.9, data_format='NCHW'
+):
+    if data_format == "AnyLayout":
+        data_format = "NCDHW"
+    return paddle.nn.functional.instance_norm(
+        input, None, None, weight, bias, True, momentum, epsilon, data_format
```
Comment (cyber-pioneer) [inline test/legacy_test/test_instance_norm_op_v2.py:32]: why fix params as None or True?

File: test/legacy_test/test_instance_norm_op_v2.py [hunk 2]
```diff
@@ -12,15 +12,69 @@
...
+    )
+
+
+def _reference_instance_norm(x, scale, bias, epsilon):
+    N, C, H, W = x.shape
```
Comment (cyber-pioneer) [inline test/legacy_test/test_instance_norm_op_v2.py:37]: how about case like layout NHWC?

File: test/legacy_test/test_instance_norm_op_v2.py [hunk 3]
```diff
@@ -12,15 +12,69 @@
...
+from paddle import fluid
+from paddle.fluid import Program, core, program_guard
+
+
+def instance_norm_wrapper(
+    input, weight, bias, epsilon=1e-5, momentum=0.9, data_format='NCHW'
+):
+    if data_format == "AnyLayout":
+        data_format = "NCDHW"
+    return paddle.nn.functional.instance_norm(
```
Comment (cyber-pioneer) [inline test/legacy_test/test_instance_norm_op_v2.py:31]: why replace paddle._C_ops.instance_norm with F.instance_norm?

File: test/legacy_test/test_instance_norm_op_v2.py (hunk 1 above)
Comment (Charles-hit) [reply test/legacy_test/test_instance_norm_op_v2.py:32]: It not be used in this api.

File: test/legacy_test/test_instance_norm_op_v2.py (hunk 2 above)
Comment (Charles-hit) [reply test/legacy_test/test_instance_norm_op_v2.py:37]: not support

File: test/legacy_test/test_instance_norm_op_v2.py (hunk 3 above)
Comment (Charles-hit) [reply test/legacy_test/test_instance_norm_op_v2.py:31]: it can't run to_static process.

**Review by cyber-pioneer — APPROVED** [review]

**Pull request merged**